File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        res = []
        for i in range(len(nums)): #0~3
            tmp = 1
            for j in range(len(nums)):
                if j == i: continue #j=0 i=1
                tmp *= nums[j]
            res.append(tmp)
        return res
            
            

File: Data Structures & Algorithms/single-number/submission-0.cpp
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        // XOR: a ^ a = 0 / a^0 = a
        int res=0;
        for (int n: nums) {
            res = res ^ n;
        }
        return res;
        //323: 011 010 011
        //res: 000->011->001->010 = 2

    }
};
